### DISCUSSION TITLE
Is it possible to generate PQC sig private key without public key in the output?

### DISCUSSION THREAD

**Discussion by xiangfeima** [open-quantum-safe/openssl#440] [Q&A]
Hi there,
I'm using below command to generate key/cert for testing
`/work/oqs-openssl/apps/openssl req -newkey dilithium2 -nodes -keyout ca.key -x509 -days 3650 -subj "/CN=my test ca" -config /work/oqs-openssl/apps/openssl.cnf -out ca.cert`
however the it looks the public key data is attached to the private key output `ca.key`.

according to https://datatracker.ietf.org/doc/html/draft-uni-qsckeys-00.html#section-7.3 the public key data is optional, so I wondering if there is any commandline option to skip the public key data part in the private key output?

thank you very much!

**Comment by dstebila** [comment]
We follow the key format specified in the NIST submissions, not the key format given in the uni-qsckeys draft. So no, we do not have such an option.

**Reply by xiangfeima** [reply]
Thank you.
then may I know is there a way to use oqs-openssl generated dilithium2 private key for liboqs signature signing interface?
https://github.com/open-quantum-safe/liboqs/blob/871f9e26d1763f512255f7e734f7a126638063c6/src/sig/sig.h#L271
Looks liboqs just requires the raw private key.

**Reply by dstebila** [reply]
The contents of the oqs-openssl private key should be the same as needed for liboqs, but they might be wrapped in a PEM or DER encoding depending on the command used to output the key, in which case you would need to extract the unencoded binary key.

**Reply by xiangfeima** [reply]
I tried ASN.1 decoding and then stripping the public part, the extracted raw private key works for liboqs, thank you!

**Comment by baentsch** [comment]
> Hi there, I'm using below command to generate key/cert for testing `/work/oqs-openssl/apps/openssl req -newkey dilithium2 -nodes -keyout ca.key -x509 -days 3650 -subj "/CN=my test ca" -config /work/oqs-openssl/apps/openssl.cnf -out ca.cert` however the it looks the public key data is attached to the private key output `ca.key`.
> 
> according to https://datatracker.ietf.org/doc/html/draft-uni-qsckeys-00.html#section-7.3 the public key data is optional, so I wondering if there is any commandline option to skip the public key data part in the private key output?
> 
> thank you very much!

You may want to try the above using OpenSSL3&oqsprovider: This is more current code that also optionally supports the unikey Draft. See How to activate here: https://github.com/open-quantum-safe/oqs-provider/blob/main/ALGORITHMS.md#key-encodings

**Reply by xiangfeima** [reply]
Thank you!, I will look into the oqs-provider for OpenSSL3.
